api/index.py

```
@@ -290,6 +432,85 @@ async def list_commits(data: RepoRequest, _=Depends(public_or_auth)):
+
+    # Validate HMAC-SHA256 signature when a secret is configured
+    if WEBHOOK_SECRET:
+        sig_header = request.headers.get("X-Hub-Signature-256", "")
+        mac = hmac.new(WEBHOOK_SECRET.encode(), body, hashlib.sha256)
+        expected_sig = "sha256=" + mac.hexdigest()
+        if not hmac.compare_digest(sig_header, expected_sig):
+            raise HTTPException(status_code=401, detail="Invalid webhook signature")
+
+    try:
+        payload = await request.json()
+    except Exception:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)